crates/syntax/src/ast/expr_ext.rs

```diff
@@ -22,6 +22,18 @@ impl ast::Expr {
             _ => false,
         }
     }
+
+    pub fn name_ref(&self) -> Option<ast::NameRef> {
```

Comment:
The same thing here -- this methong is non-primitives -- it's a convenience function which reaches deep into the node. We *should* have such functions, but not as inherent methods on AST nodes.

Reply:
the same -- see my recent comment on another PR :-)